src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -187,80 +194,96 @@ protected override bool Receive(object message)
                     Commit(commit.Offsets, msg => replyTo.Tell(msg));
                     return true;
-                
+
                 case Internal.Poll<K, V> poll:
                     ReceivePoll(poll);
                     return true;
-                
+
                 case KafkaConsumerActorMetadata.Internal.ISubscriptionRequest subscribe:
                     HandleSubscription(subscribe);
                     return true;
-                
+
+                case KafkaConsumerActorMetadata.Internal.RegisterSubStage subStage:
+                    _stageActorsMap = _stageActorsMap.SetItem(subStage.TopicPartitions, Sender);
+                    return true;
+
                 case KafkaConsumerActorMetadata.Internal.RequestMessages requestMessages:
-                    if(_settings.VerboseLogging)
-                        _log.Debug("Messages was requested, RequestId: {0}, Partitions: {1}", requestMessages.RequestId, string.Join(", ", requestMessages.Topics));
+                    if (_settings.VerboseLogging)
+                        _log.Debug("Messages was requested, RequestId: {0}, Partitions: {1}", requestMessages.RequestId,
+                            string.Join(", ", requestMessages.Topics));
                     Context.Watch(Sender);
                     CheckOverlappingRequests("RequestMessages", Sender, requestMessages.Topics);
-                    _requests = _requests.SetItem(Sender, requestMessages);
-                    _requestors = _requestors.Add(Sender);
-                    
+
+                    if (_stageActorsMap.GetOrElse(requestMessages.Topics, Sender).Equals(Sender))
+                        _requests = _requests.SetItem(Sender, requestMessages);
+
                     // When many requestors, e.g. many partitions with committablePartitionedSource the
                     // performance is much by collecting more requests/commits before performing the poll.
                     // That is done by sending a message to self, and thereby collect pending messages in mailbox.
-                    if (_requestors.Count == 1)
+                    if (_stageActorsMap.Count == 1)
                     {
                         Poll();
                     }
-                    else if (!_delayedPollInFlight)
+                    else
                     {
-                        _delayedPollInFlight = true;
-                        Self.Tell(_delayedPollMessage);
+                        RequestDelayedPoll();
                     }
+
                     return true;
-                
+
                 case KafkaConsumerActorMetadata.Internal.Seek seek:
-                    foreach (var offset in seek.Offsets)
+                    try
                     {
-                        _seekedOffset = _seekedOffset.SetItem(offset.TopicPartition, offset);
+                        foreach (var offset in seek.Offsets)
+                        {
+                            _consumer.Seek(offset);
```

Comment:
Undid some of the changes in #225 here because the way `_seekedOffset` worked was vulnerable to a race condition that could cause the `PlainPartitionSourceWithOffset` to hang. 

I think the reason why `Seek` could fail in the first place was because the partitions that were being requested were not yet assigned to the `_consumer` requesting them. Either that or the `Seek` was called too soon after the initial assignment, ala https://github.com/confluentinc/confluent-kafka-dotnet/issues/1303 (this is my guess)

I think the changes we've made to the `KafkaConsumerActor` largely alleviate the need for this work-around - might have to run a benchmark to be certain.